### PR TITLE
DEV-4817: net: sfp: workaround LOS state change miss

### PR DESCRIPTION
**Pull request by baruchsiach**
When LOS goes down (link up) read of LOS state from the edge interrupt handler sometimes return 1 (up). This makes the SFP state machine to randomly miss link up events.

Add a delay before state read to make sure we read the correct LOS state.

Reference: DEV-4817

### CODE REVIEW

**Comment by dima1308** [comment]
are we using Kernel 5.10 or 5.8?

**Comment by baruchsiach** [comment]
> are we using Kernel 5.10 or 5.8?

We current have kernel v5.8 in master. The bug only shows with kernel v5.10, probably due to some timing change in interrupts delivery. I suggest to fix this for v5.10 so we can use v5.10 in master.

File: drivers/net/phy/sfp.c [hunk 1]
```diff
@@ -318,6 +320,15 @@ static unsigned int sfp_gpio_get_state(struct sfp *sfp)
 			state |= BIT(i);
 	}
 
+	/* Trivial debounce. When no state change is detected, wait for up to
+	 * a limited bound time interval for the signal state to settle.
+	 */
+	if (state == sfp->state && repeat > 0) {
```
Comment (shmuelhazan) [inline drivers/net/phy/sfp.c:326]: What do mainline developers say about this issue? Is this workaround acceptable?

**Comment by baruchsiach** [comment]
I updated the fix with a patch sent upstream:

https://lore.kernel.org/netdev/YyA1sfZRuoka8yhl@shell.armlinux.org.uk/T/#t

I'll send an updated patch tomorrow to address Russell's comment.

File: drivers/net/phy/sfp.c (hunk 1 above)
Comment (baruchsiach) [reply drivers/net/phy/sfp.c:326]: See https://lore.kernel.org/netdev/YyA1sfZRuoka8yhl@shell.armlinux.org.uk/T/#t .

I think Russell's comment should not be too hard to address.

**Review by shmuelhazan — APPROVED** [review]

**Pull request merged**